### DISCUSSION TITLE
Inertia.js - Frontend Import Aliases

### DISCUSSION THREAD

**Discussion by akatora28** [adonisjs/core#4448] [Ideas]
In the official `@adonisjs/inertia` package, the `resources/tsconfig.json` is setup to support import aliases, `@` and `~`, however, for those import aliases to work, the `vite.config.ts` needs to include those aliasa in the `resolve` object like such:

```ts
export default defineConfig({
  plugins: [
    inertia({ ssr: { enabled: false } }),
    // frontendFramework()
    adonisjs({ entrypoints: ['resources/app.ts'], reload: ['resources/views/**/*.edge'] }),
  ],
  resolve: {
    alias: {
      '@/': `${resolve(getDirname(import.meta.url), 'resources')}/`,
      '~/': `${resolve(getDirname(import.meta.url), '.')}/`,
    },
  },
})
```

It's easy enough to add that to the config file, but I was wondering if there was any reason that couldn't (or shouldn't) be added into the official `@adonisjs/inertia` config setup? Taking a quick look under the hood, it seems like changes would need to be made to:  
[https://github.com/adonisjs/assembler/blob/develop/src/code_transformer/main.ts#L372-427](https://github.com/adonisjs/assembler/blob/develop/src/code_transformer/main.ts#L372-427)

[https://github.com/adonisjs/core/blob/df6466d330175a24cab5f0cb3fbf27f5d88a74a7/modules/ace/codemods.ts#L226-L247](https://github.com/adonisjs/core/blob/df6466d330175a24cab5f0cb3fbf27f5d88a74a7/modules/ace/codemods.ts#L226-L247)

Either those methods could be updated to add support for modifying other parts of the `vite.config.ts` file, or new functions specific to the `resolve` option could be created, then once those are updated inside the configure for the inertia package an option to update the frontend import aliases could be added in too.

Happy to take a stab at PRs on those if that's something that you think would be useful, but if the preferred option is to manually update the `vite.config.ts` understand that as well.

**Comment by Julien-R44** [comment]
hey, thanks for the feedback.
i fixed that yesterday. alias is now pre-configured when using the inertia starter kit : 

See: https://github.com/adonisjs/inertia-starter-kit/blob/main/vite.config.ts#L12

also tweaked a bit the shipped tsconfig. See this one for example
https://github.com/adonisjs/inertia/blob/main/stubs/solid/tsconfig.json.stub#L12

**Reply by akatora28** [reply]
You guys are shipping so fast, I can't keep up! Looks great though, thanks @Julien-R44 loving working with the new inertia package too!
